### PR TITLE
Added support for single source json file

### PR DESCRIPTION
**Pull request by ramamaddodi**
Update readme file for single file source json

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -63,6 +63,18 @@ node mynode.lab.local {
...
 ~~~  
 
+Second basic example, using username/password with single source json file:
+
+~~~puppet
+node mynode.lab.local {
+  class sumo {
+    email                     => 'user@example.com',
+    password                  => 'usersPassword123!', 
+    syncsourceswithsinglejson => 'PathToJsonFile/SingleSources.json',
```
Comment (ghoneycutt) [inline README.md]: Shouldn't this be a fully qualified path? Or is it relative to a a certain directory?

File: manifests/conf.pp [hunk 2]
```diff
@@ -2,23 +2,26 @@
 #
 # This class manages the sumo.conf file, which the collector uses to perform
 # initial setup.
+# Define syncsourceswithsinglejson in your class file or hiera if you want to manage single json source file.
```
Comment (ghoneycutt) [inline manifests/conf.pp]: Please remove these lines.

File: manifests/conf.pp [hunk 3]
```diff
@@ -50,9 +53,13 @@
       purge   => $syncsources_purge,
       recurse => $syncsources_recurse,
     }
-    $syncsourceswithtrailingslash = "${syncsources}/"
-  } else {
-    $syncsourceswithtrailingslash = undef
+  }
+
+  if ( $syncsourceswithsinglejson != undef)
```
Comment (ghoneycutt) [inline manifests/conf.pp]: style

```puppet
if () {
  # code
} else {
  # code
}
```

**Review by ghoneycutt — CHANGES_REQUESTED** [review]

**Comment by ghoneycutt** [comment]
I'm addressing the TravisCI issues with PR #13

**Comment by ghoneycutt** [comment]
@ramamaddodi PR #13 was merged. After you address the issues/questions above, please rebase against master and it will pick up the changes needed to allow Travis to pass.

**Comment by pjorg** [comment]
I don't know, is it really necessary to allow sources to be defined in a single JSON? The whole point of this module is to abstract the source definitions into the Puppet manifest. If you just want to write the JSON yourself you could just hard code the JSON path into your `sumo.conf` file.

I'm inclined not to include this unless a case can be made.

File: README.md (hunk 1 above)
Comment (ramamaddodi) [reply README.md]: @ghoneycutt : This need to be fully qualified path followed by filename.json. For example: '/etc/source.json'

File: manifests/conf.pp (hunk 2 above)
Comment (ramamaddodi) [reply manifests/conf.pp]: Done

File: manifests/conf.pp (hunk 3 above)
Comment (ramamaddodi) [reply manifests/conf.pp]: Fixed

**Review by ramamaddodi — COMMENTED** [review]
@ghoneycutt: Made necessary change as requested.

**Comment by ramamaddodi** [comment]
Hi @pjorg : Whole point of doing this is to give users option to choose if they want to have a single json file defining all source or individual json file for each source. Also having option of single json file eliminates defining local file resource in puppet class file for each source. This is a good option for users who have sumo collectors with more than 20-25 source type defined per environment, which can be exported as single json file from Sumo UI and imported as template, which is still manage via puppet. 
Let me know if you need more info from my end.

File: README.md [hunk 4]
```diff
@@ -63,6 +63,18 @@ node mynode.lab.local {
...
 ~~~  
 
+Second basic example, using username/password with single source json file:
+
+~~~puppet
+node mynode.lab.local {
+  class sumo {
+    email                     => 'user@example.com',
+    password                  => 'usersPassword123!', 
+    syncsourceswithsinglejson => 'PathToSingleJsonFileFollowedBySource.json',
```
Comment (ghoneycutt) [inline README.md]: perhaps `/path/to/file.json` to show it is an absolute path

File: manifests/conf.pp [hunk 5]
```diff
@@ -50,9 +51,12 @@
       purge   => $syncsources_purge,
       recurse => $syncsources_recurse,
     }
-    $syncsourceswithtrailingslash = "${syncsources}/"
-  } else {
-    $syncsourceswithtrailingslash = undef
+  }
+
+  if ( $syncsourceswithsinglejson != undef) {
```
Comment (ghoneycutt) [inline manifests/conf.pp]: if not undef, we should validate with `validate_absolute_path()`

**Comment by ghoneycutt** [comment]
@ramamaddodi if you rebase this against master it should pass tests.

**Comment by ramamaddodi** [comment]
Thank You @ghoneycutt

**Pull request merged**